protocol/storage/miner_test.go

```diff
@@ -297,18 +304,319 @@ func TestOnCommitmentAddedToChain(t *testing.T) {
 	})
 }
 
+func TestOnNewHeaviestTipSet(t *testing.T) {
+	tf.UnitTest(t)
+
+	cidGetter := types.NewCidForTestGetter()
+	proposalCid := cidGetter()
+
+	sector := testSectorMetadata(proposalCid)
+
+	t.Run("Errors if miner cannot get bootstrap miner flag", func(t *testing.T) {
+		// create new miner with deal in the accepted state and mapped to a sector
+		api, miner, _, logger := minerWithAcceptedDealTestSetup(t, proposalCid, sector.SectorID)
+
+		// return true, indicating this miner is a bootstrap miner
+		api.messageHandlers["isBootstrapMiner"] = func(a address.Address, v types.AttoFIL, p ...interface{}) ([][]byte, error) {
+			return [][]byte{}, errors.New("test error")
+		}
+
+		miner.OnNewHeaviestTipSet(types.TipSet{})
+
+		require.Equal(t, 1, len(logger.ErrorMessages))
+		assert.Contains(t, logger.ErrorMessages[0], "bootstrapping")
```

Comment:
SUGGESTION: If your intent is to exercise the invariant-checking code, I suggest that you modify `OnNewHeaviestTipSet` to return `error` (it's a void method now). Then, you can make assertions on the existence and message contained within any error returned from said method.

Reply:
Good suggestion. Done.